### ISSUE TITLE
T2T model's loss method calculates incorrect values?

### ISSUE THREAD

**Issue by AdrianLsk** [tensorflow/tensor2tensor#630]
Hi, I have a question regarding this piece of code in the T2T model class:
```python
  def loss(self, logits, features):
    if isinstance(logits, dict):
      losses = {}
      for k, v in logits.iteritems():
        losses[k] = self._loss_single(v, features)
      return tf.add_n([n / d for n, d in logits.values()])
    else:
      return self._loss_single(logits, features) 
```
So if the `logits` argument is a `dict`, then the return value is a sum of the `logit` values instead of `losses` dictionary values. But the method `self._loss_single` should output the numerator and denumerator.

Is this correct? 

For the corresponding lines of code see https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/t2t_model.py#L335-L336

**Comment by OccamsRazorResearch** [comment]
Nice catch, thanks Adrian!

**Comment by aidangomez** [comment]
Nice catch, thanks Adrian!
